### PR TITLE
chore(travis): Removed global environment variables usage.

### PR DESCRIPTION
**Pull request by robertvolkmann**
## Changes

* Moved syntax check and static code analysis in separate steps.
* Merged docker tests into one job to build the image only once.
* Removed not working phpunit update.
* Removed already installed packages.
* Split make all test into two steps.

## How to test

Compare the logs of travis-ci.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -43,58 +43,46 @@ services: postgresql
...
+        - src/testing/docker/test-cluster.sh
+        - src/testing/docker/test-standalone.sh
+      after_script: skip
   ## C/C++ agent tests ##
-    - env: CC=gcc-4.8 CXX=g++-4.8 CFLAGS='-Wall'
+    - &compiler-tests
+      env: CC=gcc-4.8 CXX=g++-4.8 CFLAGS='-Wall'
       addons:
         apt:
           packages:
```
Comment (GMishx) [inline .travis.yml]: Consider adding `sources: ubuntu-toolchain-r-test` for gcc-4.8 as well as it may break tests in future Travis versions.

Comment (robertvolkmann) [reply .travis.yml]: Done.

File: .travis.yml [hunk 2]
```diff
@@ -43,58 +43,48 @@ services: postgresql
...
-  - if [[ ${TRAVIS_PHP_VERSION:0:3} == "7.0" ]]; then composer require --dev --no-update phpunit/phpunit ^6; fi
-  - if [[ ${TRAVIS_PHP_VERSION:0:3} == "7.1" ]]; then composer require --dev --no-update phpunit/phpunit ^7; fi
-  - if [[ ! -z "${CHECKBEFORE}" ]]; then ${CHECKBEFORE}; fi
-  - if [[ ! -z "${MAKETARGETS}" ]]; then make ${MAKETARGETS}; fi
-  - if [[ ! -z "${PHPTESTSUITE}" ]]; then src/vendor/bin/phpunit -csrc/phpunit.xml --testsuite="${PHPTESTSUITE}"; fi
-  - set +e
+  - make build-lib VERSIONFILE build-cli
+  - src/vendor/bin/phpunit -csrc/phpunit.xml --testsuite="Fossology PhpUnit Test Suite"
 
 after_script:
```
Comment (maxhbr) [inline .travis.yml]: When is this part ran? every step in the matrix either has `after_script: skip` or `<<: *compiler-tests` which implies `after_script: skip`?

Comment (robertvolkmann) [reply .travis.yml]: The `after_script` is only executed in the matrix expansion. Actually three jobs (php 7.0, 5.6 and 7.1).

See: https://travis-ci.org/fossology/fossology/jobs/421474343

Comment (GMishx) [reply .travis.yml]: It is done in the PHPUnit tests?

Comment (robertvolkmann) [reply .travis.yml]: Yes, check the end of the [travis-ci log](https://travis-ci.org/fossology/fossology/jobs/421474343).

**Pull request merged**